grading/main_compile.cpp

```diff
@@ -72,108 +72,102 @@ int main(int argc, char *argv[]) {
   std::cout << "Compiling User Code..." << std::endl;
 
   system("find . -type f -exec ls -sh {} +");
+  assert (config_json.find("testcases") != config_json.end());
 
-  // Run each COMPILATION TEST
-  nlohmann::json::iterator tc = config_json.find("testcases");
-  assert (tc != config_json.end());
-  for (unsigned int i = 1; i <= tc->size(); i++) {
+  nlohmann::json testcase_config;
+  try {
+    testcase_config = find_testcase_by_id(config_json, testcase_to_compile);
+  } catch(char* c) {
+    std::cout << c << std::endl;
```

Comment:
This should never happen?
Maybe be more clear that this is an error with that print statement?
cerr?  And write "ERROR" or something too?